### PR TITLE
WV-3087 AERONET Layers Implementation

### PR DESCRIPTION
**Pull request by christof-wittreich**
## Description
This adds four new layers from AERONET in the form of special vector layers. These vector layers render as vector points, but are colored by a continuous color palette based on each vector point's value. This is a unique type of layer specific to only the AERONET layers, and as such the implementation is very custom-fit to it.

## How To Test
1. `git checkout wv-3014-aeronet-aod500nm`
2. `npm ci`
3. `npm run build`
4. `npm run watch`
5. Open the 'Add Layers' modal and verify that AERONET layers can be searched for, seen in the Aerosol Optical Depth category, and are shown in the sidebar via the AERONET source.
6. Add either of the AOD AERONET layers and verify that the color palette is applied correctly to the vector points. Select a custom color palette and verify that the vector points correctly update to the custom palette. Do the same with either of the Angstrom Exponent AERONET layers as well.
7. Verify that toggling on and off the Inactive Sites vector points via the layer settings works properly and hides the inactive gray vector points correctly.
8. Add either of the near-realtime AERONET layers and verify that the timeline changes to be hourly skips instead of daily, and that transitioning across time correctly updates which vector points have an active value.
9. Verify that clicking on any of the vector points from any zoom level brings up a vector dialog, and that it contains the correct information and links for that site.
10. Verify that mousing-over a vector point correctly highlights the color palette on the side panel at the correct color, or the Inactive Sites vector color square on the side panel for inactive sites.
11. Briefly verify that other vector layers are not broken by the changes implemented here.

### CODE REVIEW

**Comment by PatchesMaps** [comment]
The application crashes as soon as I add an AERONET layer

File: web/js/map/layerbuilder.js [hunk 1]
```diff
@@ -479,6 +490,236 @@ export default function mapLayerBuilder(config, cache, store) {
...
+    });
+
+    let colors = [];
+    let values = [];
+    if (state.palettes.rendered[def.id]) {
+      colors = state.palettes.rendered[def.id].maps[1].entries.colors;
+      values = state.palettes.rendered[def.id].maps[1].entries.values;
+    }
+
+    const requestDate = util.toISOStringSeconds(util.roundTimeOneMinute(date)).slice(0, 10);
```
[the remark this answers is not in the document]
Comment (christof-wittreich) [reply web/js/map/layerbuilder.js:636]: We do the same process for ttiler and xyz layers (as can be seen in this same file), so it was included to match those. I don't think it's needed though, and the other vector layers do not use it, so I'll remove this.

File: web/js/map/layerbuilder.js [hunk 2]
```diff
@@ -479,6 +490,236 @@ export default function mapLayerBuilder(config, cache, store) {
...
+          const data = await res.text();
+          return data;
+        };
+
+        const getActiveData = async () => {
+          const avg = def.id.includes('DAILY') ? 20 : 10;
+          const date2 = new Date(date.toString());
+          date2.setHours(date.getHours() + 1);
+          const urlParameters = `?year=${date.getUTCFullYear()}&month=${date.getUTCMonth() + 1}&day=${date.getUTCDate()}&year2=${date2.getUTCFullYear()}&month2=${date2.getUTCMonth() + 1}&day2=${date2.getUTCDate()}${isSubdaily ? `&hour=${date.getUTCHours()}&hour2=${date2.getUTCHours()}` : ''}&AOD15=1&AVG=${avg}&if_no_html=1`;
+          const res = await fetch(source.url + urlParameters);
```
[the remark this answers is not in the document]
Comment (christof-wittreich) [reply web/js/map/layerbuilder.js:552]: @PatchesMaps line 551 here (`const urlParameters = ...`) is already formatted as a template literal. Did you have feedback on how I wrote the template literal, or was your comment intended for a different line?

[the remark this answers is not in the document]
Comment (PatchesMaps) [reply web/js/map/layerbuilder.js:552]: It was intended for line 552: `${source.url}${urlParameters}` is what I had in mind.

**Review by minniewong — APPROVED** [review]

**Review by PatchesMaps — APPROVED** [review]

**Pull request merged**